src/Serilog/Build/ImplicitDependencyAttribute.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+
+namespace Serilog.Build
+{
+    /// <summary>
+    /// Indicates that the marked assembly depends on a type that is specified in the constructor.
+    /// Typically used to force a compile-time dependency to the assembly that contains the type.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    public sealed class ImplicitDependencyAttribute : Attribute
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ImplicitDependencyAttribute"/> class.
+        /// </summary>
+        /// <param name="dependentType"></param>
+        public ImplicitDependencyAttribute(Type dependentType)
+        {
+            DependentType = dependentType;
+        }
+
+        /// <summary>
+        /// Gets the dependent type reference.
+        /// </summary>
+        /// <value>The dependent type reference.</value>
+        public Type DependentType { get; private set; }
```

Comment:
_Dependent_, to me, is the thing that "does-the-depending" (i.e. the inverse direction). Maybe `Dependency`?